salt/states/lvm.py

```diff
@@ -126,7 +126,7 @@ def vg_present(name, devices=None, **kwargs):
         ret['comment'] = 'Volume Group {0} already present'.format(name)
         for device in devices.split(','):
             realdev = os.path.realpath(device)
-            pvs = __salt__['lvm.pvdisplay'](realdev)
+            pvs = __salt__['lvm.pvdisplay'](realdev, real=True)
             if pvs and pvs.get(realdev, None):
                 if pvs[realdev]['Volume Group Name'] == name:
                     ret['comment'] = '{0}\n{1}'.format(
```

Comment:
This isn't extending the vg, that just reads which vg that device is owned by.

The problem that the original patch solves is that you can use udev rules to create any number of symlinks to devices. You can symlink /dev/sda to /dev/monkeydisk and add it as a pv. When you do, however, pvdisplay shows /dev/sda not /dev/monkeydisk.

lvm2 wrote in a special case of rules that treat /dev/mapper symlinks differently. What if they do that with other classes of devices? By using the realpaths for testing that the devices are part of the vg, we can ensure that the state is correct or not accurately.